### DISCUSSION TITLE
what is this error and why does this only occur in home network and not on vps?

### DISCUSSION THREAD

**Discussion by breezemight** [projectdiscovery/retryabledns#56] [General]
running queries in for loop pops us this error `read udp 192.168.19.131:40138->8.8.4.4:53: i/o timeout` after runing for a few seconds in home computer (which has slow internet speed). But this works perfectly fine in VPS. 
Why does this happen and how can it be solved?

**Comment by atc0005** [comment]
I haven't used this library before (considering it for a project), but saw your question and thought I'd respond.

I believe that the error message you shared is from an attempt to interact with Google's Public DNS at 8.8.4.4 via UDP which failed to get a response within a set timeout limit.

I suspect that you may see this more often from a home network than you would from a VPS provider, likely due to packet loss or latency on your connection (vs a more expensive/less congested commercial network connection used by a VPS provider).

I'll defer to those more knowledgeable, but it may be worth retrying from a TCP-based DNS query when an initial UDP-based query fails. Or, increase the number of retry attempts. I've not looked into this library sufficiently to know whether an automatic TCP-based query is offered if a UDP query fails a specific number of times.

How often do you see timeouts like the one you shared error?
